boards.txt

```diff
@@ -2800,6 +2800,479 @@ aventen_s3_sync.menu.EraseFlash.all.upload.erase_cmd=-e
 
 ##############################################################
 
+BharatPi-Node-Wifi.name=BharatPi Node Wifi Module
+
+BharatPi-Node-Wifi.bootloader.tool=esptool_py
+BharatPi-Node-Wifi.bootloader.tool.default=esptool_py
+
+BharatPi-Node-Wifi.upload.tool=esptool_py
+BharatPi-Node-Wifi.upload.tool.default=esptool_py
+BharatPi-Node-Wifi.upload.tool.network=esp_ota
+
+BharatPi-Node-Wifi.upload.maximum_size=1310720
+BharatPi-Node-Wifi.upload.maximum_data_size=327680
+BharatPi-Node-Wifi.upload.flags=
+BharatPi-Node-Wifi.upload.extra_flags=
+
+BharatPi-Node-Wifi.serial.disableDTR=true
+BharatPi-Node-Wifi.serial.disableRTS=true
+
+BharatPi-Node-Wifi.build.tarch=xtensa
+BharatPi-Node-Wifi.build.bootloader_addr=0x1000
+BharatPi-Node-Wifi.build.target=esp32
+BharatPi-Node-Wifi.build.mcu=esp32
+BharatPi-Node-Wifi.build.core=esp32
+BharatPi-Node-Wifi.build.variant=BharatPi-Node-Wifi
+BharatPi-Node-Wifi.build.board=ESP32_WROOM_DA
+
+BharatPi-Node-Wifi.build.f_cpu=240000000L
+BharatPi-Node-Wifi.build.flash_size=4MB
+BharatPi-Node-Wifi.build.flash_freq=40m
+BharatPi-Node-Wifi.build.flash_mode=dio
+BharatPi-Node-Wifi.build.boot=dio
+BharatPi-Node-Wifi.build.partitions=default
+BharatPi-Node-Wifi.build.defines=
+BharatPi-Node-Wifi.build.loop_core=
+BharatPi-Node-Wifi.build.event_core=
+
+BharatPi-Node-Wifi.menu.PartitionScheme.default=Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS)
```

Comment:
Also go through all partition schemes and remove those that are not for 4MB Flash size.